realm/src/main/java/io/realm/dynamic/DynamicRealmList.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.dynamic;
+
+import java.util.AbstractList;
+import java.util.Iterator;
+import java.util.ListIterator;
+
+import io.realm.Realm;
+import io.realm.internal.LinkView;
+
+/**
+ * RealmList exposed using a dynamic API. All objects in the list must have the same schema even though they are accessed
+ * dynamically.
+ */
+public class DynamicRealmList extends AbstractList<DynamicRealmObject> {
+
+    private final LinkView linkView;
+    private final Realm realm;
+
+    DynamicRealmList(LinkView linkView, Realm realm) {
+        this.linkView = linkView;
+        this.realm = realm;
+    }
+
+    @Override
+    public boolean add(DynamicRealmObject object) {
+        checkIsValidObject(object);
+        linkView.add(object.row.getIndex());
+        return true;
+    }
+
+    @Override
+    public void clear() {
+        linkView.clear();
+    }
+
+    @Override
+    public DynamicRealmObject get(int location) {
+        checkValidIndex(location);
+        return new DynamicRealmObject(realm, linkView.get(location));
+    }
+
+    @Override
+    public DynamicRealmObject remove(int location) {
+        DynamicRealmObject removedItem = get(location);
+        linkView.remove(location);
+        return removedItem;
+    }
+
+    @Override
+    public DynamicRealmObject set(int location, DynamicRealmObject object) {
+        checkIsValidObject(object);
+        checkValidIndex(location);
+        linkView.set(location, object.row.getIndex());
+        return object;
+    }
+
+    @Override
+    public int size() {
+        return ((Long)linkView.size()).intValue();
```

Comment:
What do you mean handle it? Even if every row only used 1 byte, a list with size = max_int would still hold 2 GB of data, so this would not be relevant for a mobile application?    

Fun fact from the Collections JavaDoc: "size(): Returns the number of elements in this collection. If this collection contains more than Integer.MAX_VALUE elements, returns Integer.MAX_VALUE."

So we should actually do a ```Math.min(linkView.size(), Integer.MAX_VALUE)``` here if want to honour that contract.